src/main.cpp

```diff
@@ -100,33 +100,33 @@ int main(int argc, const char** argv)
     LOG_INFO("{} Started", productName);
 
     try {
-        while (!unattended) {
-            char response = 'N';
-            fmt::print("{} will now modify your system, do you want to "
-                       "continue? [Y/N]\n",
-                cloyster::productName);
-            std::cin >> response;
-
-            if (response == 'Y' || response == 'y') {
-                LOG_INFO("Running {}.\n", cloyster::productName)
-                break;
-            } else if (response == 'N' || response == 'n') {
-                LOG_INFO("Stopping {}.\n", cloyster::productName);
-                return EXIT_SUCCESS;
-            }
-        }
-
         if (cloyster::showVersion) {
             fmt::print("{}: Version {}\n", productName, productVersion);
             return EXIT_SUCCESS;
         }
 
-        if (cloyster::runAsRoot) {
-            cloyster::checkEffectiveUserId();
-        }
-
         if (cloyster::dryRun) {
             fmt::print("Dry run enabled.\n");
+        } else {
+            while (!unattended) {
+                char response = 'N';
+                fmt::print("{} will now modify your system, do you want to "
+                           "continue? [Y/N]\n",
+                    cloyster::productName);
+                std::cin >> response;
+
+                if (response == 'Y' || response == 'y') {
+                    LOG_INFO("Running {}.\n", cloyster::productName)
+                    break;
+                } else if (response == 'N' || response == 'n') {
+                    LOG_INFO("Stopping {}.\n", cloyster::productName);
+                    return EXIT_SUCCESS;
+                }
+            }
+        }
+
+        if (cloyster::runAsRoot) {
```

Comment:
That should be the one of the first checks. Do not ask for changes if it will fail after.

Reply:
Why fail? It's in the same place it was before...

Reply:
Awesome the fix. Thanks.